staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go

```diff
@@ -277,6 +277,19 @@ func IsHTTPHeaderName(value string) []string {
 	return nil
 }
 
+const envVarNameFmt = "[-._a-zA-Z][-._a-zA-Z0-9]*"
```

Comment:
Yes, that's what I meant -- I was a bit sloppy, sorry.

So I understand we should stay on the safe side and make sure environment variables are subject to the same dot-specific restrictions. Will do that unless I hear objections.